src/store/modules/records/actions.js

```diff
@@ -0,0 +1,81 @@
+import debug from 'debug'
+
+import Record from '../../../util/api/record'
+import Transfer from '../../../util/api/transfer'
+
+const logger = debug('app:store:records:actions')
+
+export default {
+  FETCH_USER_DATA({ dispatch }) {
+
+    // Don't block on any of these async actions
+    // The UI will fill in as they are populated
+    dispatch('FETCH_USER_RECORDS')
+    dispatch('FETCH_INCOMING_TRANSFERS')
+    dispatch('FETCH_OUTGOING_TRANSFERS')
+  },
+
+  FETCH_USER_RECORDS({ commit }) {
+    logger('FETCH_USER_RECORDS action being executed')
+
+    return Record.getUserRecords()
+      .then((records) => {
+        commit('SET_RECORDS', {
+          listName: 'userRecords',
+          records,
+        })
+      })
+  },
+
+  FETCH_INCOMING_TRANSFERS({ commit }) {
+    logger('FETCH_INCOMING_TRANSFERS action being executed')
+
+    return Transfer.getIncomingTransfers()
+      .then((records) => {
+        commit('SET_RECORDS', {
+          listName: 'incomingTransfers',
+          records,
+        })
+      })
+  },
+
+  FETCH_OUTGOING_TRANSFERS({ commit }) {
+    logger('FETCH_OUTGOING_TRANSFERS action being executed')
+
+    return Transfer.getOutgoingTransfers()
+      .then((records) => {
+        commit('SET_RECORDS', {
+          listName: 'outgoingTransfers',
+          records,
+        })
+      })
+  },
+
+  FETCH_RECORD({ commit, state }, tokenId) {
+    logger('FETCH_RECORD action being executed', tokenId)
+
+    // First, check to see if we've cached this in the userRecords array
+    const existingRecord = state.userRecords.find((userRecord) => {
```

Comment:
Only `FETCH_RECORD` is cached? I feel like all methods should do this, and you should also be able to bust the cache. Something like this is what I was thinking:

```javascript
FETCH_RECORD({ commit, state }, { tokenId, bustCache = false }) {
  logger('FETCH_RECORD action being executed', tokenId)

  // First, check to see if we've cached this in the userRecords array
  const existingRecord = !bustCache && state.userRecords.find((userRecord) => {
    return userRecord.tokenId === tokenId
  })

  return Promise.resolve(existingRecord || Record.getRecord(tokenId))
    .then((record) => {
      commit('SET_ACTIVE_RECORD', record)
    })
},
```

But then I guess you have to get into the question of when do you bust the cache? Maybe caching everything is overkill for now.

Reply:
Discussed offline